src/com/walmartlabs/lacinia/pedestal/subscriptions.clj

```diff
@@ -114,21 +115,24 @@
              (case type
                "connection_init"
                (when (>! response-data-ch {:type :connection_ack})
-                 (recur subs))
+                 (recur subs (assoc connection-state :initial-payload payload)))
 
                 ;; TODO: Track state, don't allow start, etc. until after connection_init
 
                "start"
-               (do
+               (let [context (merge connection-state context)
```

Comment:
At pet peeve but, I prefer not to overwrite a name, so I'd probably call this either `merged-context` or `context'`.